tests/lib/resource_test.py

```diff
@@ -317,7 +337,94 @@ def test_test_entrypoint_success():
     event = resource.test_entrypoint.__wrapped__(  # pylint: disable=no-member
         resource, payload, None
     )
-    assert event is sentinel.response
+    assert event is progress_event
 
     mock_model._deserialize.assert_has_calls([call(None), call(None)])
     mock_handler.assert_called_once()
+
+
+def test_schedule_reinvocation_not_in_progress():
+    mock_progress_event = Mock(
+        "cloudformation_cli_python_lib.interface.ProgressEvent", autospec=True
+    )()
+    mock_progress_event.status = OperationStatus.SUCCESS
+    with patch(
+        "cloudformation_cli_python_lib.resource.boto3.Session", autospec=True
+    ) as mock_session, patch(
+        "cloudformation_cli_python_lib.resource.CloudWatchScheduler", autospec=True
+    ) as mock_scheduler:
+        reinvoke = Resource.schedule_reinvocation(
+            sentinel.request,
+            mock_progress_event,
+            sentinel.context,
+            sentinel.credentials,
+        )
+    assert reinvoke is False
+    mock_session.assert_not_called()
+    mock_scheduler.assert_not_called()
+
+
+def test_schedule_reinvocation_local_callback():
+    mock_progress_event = Mock(
+        "cloudformation_cli_python_lib.interface.ProgressEvent", autospec=True
+    )()
+    mock_progress_event.status = OperationStatus.IN_PROGRESS
+    mock_progress_event.callbackDelaySeconds = 5
+    mock_request = Mock(
+        "cloudformation_cli_python_lib.interface.HandlerRequest", autospec=True
+    )()
+    mock_request.requestContext = {}
+    mock_context = Mock(
+        "cloudformation_cli_python_lib.interface.LambdaContext", autospec=True
+    )()
+    mock_context.get_remaining_time_in_millis.return_value = 600000
+    with patch(
+        "cloudformation_cli_python_lib.resource.boto3.Session", autospec=True
+    ) as mock_session, patch(
+        "cloudformation_cli_python_lib.resource.CloudWatchScheduler", autospec=True
```

Comment:
if these aren't being called, won't not patching and not checking that they aren't called accomplish the same thing? im all for explicitly checking that things are not called for clarity, but this is pretty verbose. Curious what others thought; I don't have a strong opinion either way

Reply:
put this out of fear of a invocation storm if we return `True` (re-invoke locally) and then on every eval, create another cloudwatch scheduled invoke. Although looking at it again, without a valid function_arn being attached to the `mock_context` this test will catch those scenarios without the mocks and explicit checks.